### PR TITLE
Enhance QueryRunner to support single thread, multi threads and target QPS modes.

### PR DESCRIPTION
**Pull request by Jackie-Jiang**

### CODE REVIEW

File: pinot-perf/src/main/java/com/linkedin/pinot/perf/QueryRunner.java [hunk 1]
```diff
@@ -19,58 +19,122 @@
...
-  }
+  @Option(name = "-queryFile", required = true, usage = "query file path")
+  private String _queryFile;
+  @Option(name = "-mode", required = true, usage = "query runner mode (singleThread|multiThreads|targetQPS)")
+  private String _mode;
+  @Option(name = "-numThreads", required = false, usage = "number of threads for multiThread mode and targetQPS mode")
+  private int _numThreads;
+  @Option(name = "-startQPS", required = false, usage = "start QPS for targetQPS mode")
+  private double _startQPS;
+  @Option(name = "-deltaQPS", required = false, usage = "delta QPS for targetQPS mode")
```
Comment (mayankshriv) [inline pinot-perf/src/main/java/com/linkedin/pinot/perf/QueryRunner.java]: Not a requirement for this commit, but you may want to consider adding checks like mutually exclusive arguments, dependent arguments etc.

File: pinot-perf/src/main/java/com/linkedin/pinot/perf/QueryRunner.java [hunk 2]
```diff
@@ -91,92 +155,162 @@ public void run() {
     }
   }
 
-  public static void singleThreadedQueryRunner(PerfBenchmarkDriverConf conf, String queryFile)
+  /**
+   * Use multiple threads to run query at an increasing target QPS.
```
Comment (mayankshriv) [inline pinot-perf/src/main/java/com/linkedin/pinot/perf/QueryRunner.java]: Add more detailed explanation describing the algorithm, eg if/when threads wait etc.

File: pinot-perf/src/main/java/com/linkedin/pinot/perf/QueryRunner.java [hunk 3]
```diff
@@ -91,92 +155,162 @@ public void run() {
...
 
-  public static JSONObject runSingleQuery(PerfBenchmarkDriverConf conf, String query, int numRuns)
-      throws Exception {
-    PerfBenchmarkDriver driver = new PerfBenchmarkDriver(conf);
-    JSONObject response = null;
-
-    for (int i = 0; i < numRuns; i++) {
-      response = driver.postQuery(query);
+  private static void printUsage() {
+    System.out.println("Usage: QueryRunner");
```
Comment (mayankshriv) [inline pinot-perf/src/main/java/com/linkedin/pinot/perf/QueryRunner.java]: Use logger instead of System.out

**Comment by mayankshriv** [comment]
LGTM, other than the comments.

File: pinot-perf/src/main/java/com/linkedin/pinot/perf/QueryRunner.java (hunk 3 above)
Comment (Jackie-Jiang) [inline pinot-perf/src/main/java/com/linkedin/pinot/perf/QueryRunner.java]: Use System.out just for command line interaction.

**Pull request merged**